relational_operators/SelectOperator.cpp

```diff
@@ -54,43 +54,93 @@ bool SelectOperator::getAllWorkOrders(
 
   if (input_relation_is_stored_) {
     if (!started_) {
-      for (const block_id input_block_id : input_relation_block_ids_) {
-        container->addNormalWorkOrder(
-            new SelectWorkOrder(input_relation_,
-                                input_block_id,
-                                predicate,
-                                simple_projection_,
-                                simple_selection_,
-                                selection,
-                                output_destination,
-                                storage_manager),
-            op_index_);
+      if (input_relation_.hasPartitionScheme()) {
+        const PartitionScheme &part_scheme =
+            input_relation_.getPartitionScheme();
+        int num_partitions = part_scheme.getPartitionSchemeHeader().getNumPartitions();
```

Comment:
Please mark `num_partitions  const`.

Actually, we could refactor it as the following, since you don't use `part_scheme` elsewhere.

```
  const std::size_t num_partitions =
      input_relation_.getPartitionScheme().getPartitionSchemeHeader().getNumPartitions();
```